### ISSUE TITLE
Printing to console through cy.task("log", "Hello World") before cy.visit causes it to print twice

### ISSUE THREAD

**Issue by alexsch01** [cypress-io/cypress#24720]
### Current behavior

Printing out to the console through cy.task
https://docs.cypress.io/api/commands/task#Syntax:~:text=cy.task(%27log%27%2C%20%27This%20will%20be%20output%20to%20the%20terminal%27) before doing a cy.visit causes the printout to occur twice

cy.task("log", "Hello World") would cause the following.....

Hello World # happens almost right away as it should
Hello World # occurs just before the page loads
[page loads]

### Test code to reproduce
["cypress.config.js"]
```
const { defineConfig } = require('cypress')

module.exports = defineConfig({
  // setupNodeEvents can be defined in either
  // the e2e or component configuration
  e2e: {
    setupNodeEvents(on, config) {
      on('task', {
        log(message) {
          console.log(message)
          return null
        },
      })
    }
  }
})
```

["cypress/e2e/spec.cy.js" from Create an Empty Spec with cy.task part added]
```
describe('empty spec', () => {
  it('passes', () => {
    cy.task("log", "Hello World")
    cy.visit('https://example.cypress.io')
  })
})
```
[Every other cypress folder/file not modified]

### Cypress Version

11.1.0

### Node version

18.12.1

### Operating System

Debian 11.5, also tested in Windows 10.0.19045.2251

**Comment by alexsch01** [comment]
This is when running the command: "npx cypress run --spec cypress/e2e/spec.cy.js"

**Comment by alexsch01** [comment]
Video of the bug:
[Screen recording 2022-11-18 10.48.08 AM.webm](https://user-images.githubusercontent.com/5721147/202745410-0fe3f33a-9d72-40e9-a111-76ed28dc5da3.webm)

**Comment by marktnoonan** [comment]
Thanks for reporting @alexsch01 - the reason for this is that when you `cy.visit()` a new domain, Cypress has to switch to that domain. This causes Cypress to refresh with a new URL and the spec restarts.

If you set your `baseUrl` in config to `https://example.cypress.io/`, Cypress will be aware of that when it opens, and not need to switch when the spec does its first `visit`.

Out of curiosity, what is the impact for of this for you? Are you wanting to so setup before `cy.visit` in an end to end test?

**Comment by alexsch01** [comment]
Where I work, (company protocol) we print a specific text format (with things like environment of the application, test name, etc) to the console before the spec runs.

What's rather interesting is that this workaround works and only prints "Hello World" once before any cy.visit's happen
```
describe('empty spec', () => {
  it(' ', () => {
    cy.task('log', "Hello World");
  });

  it('passes', () => {
    cy.visit('https://example.cypress.io')
  })
})
```

**Comment by marktnoonan** [comment]
Hmm. I don't know too much about this area, it looks like we don't re-run the whole spec then, we just pick up at the it block after the domain changes.

What you're describing seems like something that might fit better in a `before` or `beforeEach` hook, is there a reason you prefer to do this inside the `it` block?

**Comment by alexsch01** [comment]
The reason I used an it block is that the below code has the same issue with printing "Hello World" twice.

```
describe('empty spec', () => {
  before(() => {
    cy.task('log', "Hello World");
  });

  it('passes', () => {
    cy.visit('https://example.cypress.io')
  })
})
```

**Comment by alexsch01** [comment]
Any update on this @marktnoonan ?

Functionally, this ugly code (https://github.com/cypress-io/cypress/issues/24720#issuecomment-1320269694) works - just doesn't look that great

Thank you for your time

**Comment by marktnoonan** [comment]
I did look into this a bit more, I was surprised when I saw your comment about the `before` block. And somebody shared this issue with me: https://github.com/cypress-io/cypress/issues/2777

This confirms it's currently the expected behavior for Cypress to behave this way, I'll close this issue in favor of continuing to track in that main issue. I do agree this is not intuitive. 

There are some other workarounds in that thread it looks like, although I think yours might be the best for your situation for the time being.
